File: 2857-total-distance-traveled/total-distance-traveled.java
class Solution {
    public int distanceTraveled(int mainTank, int additionalTank) {
     int r=0;
     for(int i=1;i<=mainTank;i++){
        if(additionalTank!=0&&i%5==0){
            additionalTank-=1;
            r+=10;
            mainTank+=1;
        }
        else{
            r+=10;
        }
     }
     return r;   
    }
}

File: 2857-total-distance-traveled/total-distance-traveled.c
int distanceTraveled(int mainTank, int additionalTank) {
    int d=0;
     while(mainTank>0){
        if(mainTank>=5&&additionalTank>0){
            d+=5;
            mainTank=mainTank-4;
            additionalTank--;        
        }else{
            d=(d+mainTank)*10;
            mainTank=0;
        }
     }
     return d;
}